tests/all_tests.sh

```diff
@@ -1,13 +1,9 @@
 #!/bin/bash
 
-srcdir="$(cd "$(dirname "${BASH_SOURCE[0]}")"/.. && pwd)"
+cd "$(dirname "$0")"/..
```

Comment:
The original script didn't change directory because the output directory, as well as the path to the generated pycdc executable, was relative to the build path...  This version of the script no longer works with out-of-tree builds.

Reply:
I'm not sure what you refer to as "out-of-tree builds". 

`$0` refers to the program name called from the command line, e.g. `./tests/all-test.sh` or `/absolute/path/tests/all-tests.sh`
`dirname $0` will get the parent directory, which we use as a base to change current working directory. The change of directory is only effective inside the subshell of the script (and all its children). The underlying script `decompyle_test.sh` will use relative paths from that.

Maybe I am missing one scenario where this fails, are you able to develop on this?

Reply:
out-of-tree builds mean when the build tree is not directly in the source tree (which is generally the preferred way of using cmake).  For example, see the GitHub workflows, which do the build in a /build directory.

This script doesn't directly deal with the path to the built pycdc executable, but the called script (decompyle_test.sh) does, and it currently assumes that pycdc is in the cwd.

Reply:
Alright, thanks for taking the time to explain the issue - I do see it now and it is a terrible mistake. I've opted for a similar approach that was in the original script.
I also chose to squash all the commits as they were part of the review process.
Cheers

Reply:
I think I now understand why you use this construction:
```
srcdir="$(cd "$(dirname "${BASH_SOURCE[0]}")"/.. && pwd)"
```
Instead of a simple
```
srcdir="$(dirname "$0")"
```
_Thank you Windows back slashes_
Fix on the way.